broker/src/main/java/org/apache/rocketmq/broker/processor/PopMessageProcessor.java

```diff
@@ -734,15 +767,13 @@ private int polling(final Channel channel, RemotingCommand remotingCommand,
         if (requestHeader.getPollTime() <= 0 || this.popLongPollingService.isStopped()) {
             return NOT_POLLING;
         }
-        ConcurrentHashMap<String, Byte> cids = topicCidMap.get(requestHeader.getTopic());
-        if (cids == null) {
-            cids = new ConcurrentHashMap<>();
-            ConcurrentHashMap<String, Byte> old = topicCidMap.putIfAbsent(requestHeader.getTopic(), cids);
-            if (old != null) {
-                cids = old;
-            }
-        }
-        cids.putIfAbsent(requestHeader.getConsumerGroup(), Byte.MIN_VALUE);
+        topicCidMap.computeIfAbsent(requestHeader.getTopic(), k -> new ConcurrentHashMap<>())
+            .putIfAbsent(requestHeader.getConsumerGroup(), Byte.MIN_VALUE);
+        final String popRetryTopic
+            = KeyBuilder.buildPopRetryTopic(requestHeader.getTopic(), requestHeader.getConsumerGroup());
+        topicCidMap.computeIfAbsent(popRetryTopic, k -> new ConcurrentHashMap<>())
```

Comment:
Why we need put popRetryTopic into topicCidMap? In notifyRetryMessageArriving, the topic has been changed to normal topic.

Reply:
Because `notifyMessageArriving(final String toic, final int queueId)` is called by ReputMessageService, the topic won't be changed to normal topic.
![image](https://user-images.githubusercontent.com/10664298/236685934-e797e1ae-bbf9-4e6e-9d0a-c6d377d1f43a.png)

Reply:
But I didn't notice `notifyMessageArriving` is called in `PopReviveService#reviveRetry`. 
It looks like it would be more appropriate to call `notifyRetryMessageArriving` in `PopReviveService#reviveRetry`.